engine.js

```diff
@@ -160,22 +161,26 @@ var engine = function(cloudConfig, settings) {
                     // Add this to our tracking for the worst status to calculate
                     // the exit code
                     maximumStatus = Math.max(maximumStatus, results[r].status);
-                    // Remediation
+                    if (results[r].status === 2 && failedResources.indexOf(results[r].resource) === -1){
+                        failedResources.push(results[r].resource);
+                    }
+                }
+                async.forEach( failedResources,(resource, rcb) => {
                     if (settings.remediate && settings.remediate.length) {
```

Comment:
you should check `if (settings.remediate && settings.remediate.length) {` before going through the async loop, that way if the remediation is not set it won't even loop through those resources